mongodbatlas/provider.go

```diff
@@ -799,3 +800,17 @@ func expandAssumeRole(tfMap map[string]interface{}) *AssumeRole {
 func pointer[T any](x T) *T {
 	return &x
 }
+
+func intPtr(v int) *int {
+	if v != 0 {
+		return &v
+	}
+	return nil
+}
+
+func stringPtr(v string) *string {
+	if v != "" {
+		return &v
+	}
+	return nil
+}
```

Comment:
Why you need to define these function instead of using the one at line 800 (`pointer`)?

Reply:
`pointer` function only checks whether the passed attribute is nil or not, we need to add additional checks here for 0s and empty strings because of the issue related to default values in Atlas and the workaround suggested by @AgustinBettati in Slack (shared). Essentially, we're not sending empty strings in the API requests.

Reply:
thanks for the clarification